File: python/68_letter_43B.py
from collections import Counter
import sys

lines = sys.stdin.read().splitlines()
if len(lines) < 2:
    sys.exit(0)
s1 = lines[0]
s2 = lines[1]
freq = Counter(c for c in s1 if c != " ")
for c in s2:
    if c == " ":
        continue
    if freq[c] <= 0:
        print("NO")
        break
    freq[c] -= 1
else:
    print("YES")

File: cpp/68_letter_43B.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string s1, s2;
    if (!getline(cin, s1))
        return 0;
    if (!getline(cin, s2))
        return 0;
    map<char, int> freq;
    for (char c : s1)
        if (c != ' ')
            freq[c]++;
    for (char c : s2)
        if (c != ' ')
        {
            if (freq[c] == 0)
            {
                cout << "NO\n";
                return 0;
            }
            freq[c]--;
        }
    cout << "YES\n";
}
